src/bin/communication_thread.rs

```diff
@@ -0,0 +1,65 @@
+//! Example on how to use a communication thread alongside with the GUI thread.
+//! 
+//! Tricks used here:
+//! - Use a queue to show data on the GUI.
+//! - Run an async function on the GUI event loop.
+//! - Use a seperate thread to handle incoming data and put it into a queue.
+
+extern crate futures;
+extern crate gio;
+extern crate glib;
+extern crate gtk;
+use futures::StreamExt;
+use gio::prelude::*;
+use gtk::prelude::*;
+use gtk::{ApplicationWindow, Label};
+use std::env::args;
+use std::thread;
+
+fn main() {
+    let application = gtk::Application::new(
+        Some("com.github.gtk-rs.examples.communication_thread"),
+        Default::default(),
+    )
+    .expect("Initialization failed...");
+    application.connect_activate(build_ui);
+    application.run(&args().collect::<Vec<_>>());
+}
+
+fn build_ui(application: &gtk::Application) {
+    let window = ApplicationWindow::new(application);
+    let label = Label::new(None);
+    window.add(&label);
+
+    // Create a channel between communication thread and main event loop:
+    let (sender, mut receiver) = futures::channel::mpsc::channel::<String>(1000);
```

Comment:
```suggestion
    let (sender, mut receiver) = futures::channel::mpsc::channel(1000);
```

The type should be possible to infer or not? :)

Reply:
It infers `str` which does not have a size at compile time.

Reply:
Curious, I would've thought that `start_communication_thread()` would've made the compiler know that it must be a `String`

Reply:
Me too, but I assume the type is inferred from the first usage, which is in the `set_text` function, so then it is determined at `str` instead of `String`.

Reply:
Oh well :)